swagger2.0/oic.r.vodlist.swagger.json

```diff
@@ -0,0 +1,135 @@
+                      "econame": "Z-Wave"
+                    },
+                    {
+                      "n": "Door Lock",
+                      "di": "54919CA5-4101-4AE4-595B-353C51AA5678",
+                      "econame": "Zigbee"
+                    }
+                  ]
+                },
+              "schema": { "$ref": "#/definitions/vodlist" }
+            }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "interface" : {
+      "in" : "query",
+      "name" : "if",
+      "type" : "string",
+      "enum" : ["oic.if.r", "oic.if.baseline"]
+    }
+  },
+  "definitions": {
+    "vodentry" : {
+      "description": "Information of a VOD created by the Bridge",
+      "type": "object",
+      "properties": {
+        "n": {
+            "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.common.properties.core-schema.json#/definitions/n"
+        },
+        "di" : {
+          "description": "Device Id of VOD",
+          "allOf": [
+            {
+              "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.types-schema.json#/definitions/uuid"
+            },
+            {
+              "description": "Unique identifier for device",
+              "readOnly": true
+            }
+          ]
+        },
+        "econame": {
+          "description": "Ecosystem Name of the Bridged Device which is exposed by this VOD",
+          "type": "string",
+          "enum": [ "BLE", "oneM2M", "U+", "Zigbee", "Z-Wave" ],
+          "readOnly": true
+        }
+      },
+      "required": ["n", "di", "econame"]
+    },
+    "vodlist": {
+      "type": "object",
+      "properties": {
+        "n" :   {
+          "description": "Friendly name of the resource",
+          "maxLength": 64,
+          "readOnly": true,
+          "type": "string"
+        },
+        "rt" :  {
+          "description": "Resource Type",
+          "items": {
+            "maxLength": 64,
+            "type": "string",
+            "enum": ["oic.r.vodlist"]
+          },
+          "minItems": 1,
+          "readOnly": true,
+          "type": "array"
+        },
+        "id" :   {
+          "description": "Instance ID of this specific resource",
+          "maxLength": 64,
+          "readOnly": true,
+          "type": "string"
+        },
+        "if" :  {
+          "description": "The interface set supported by this resource",
+          "items": {
+            "enum": [
+              "oic.if.baseline",
+              "oic.if.r"
+            ],
+            "type": "string"
+          },
+          "minItems": 1,
+          "readOnly": true,
+          "type": "array"
+        },
+        "vods": {
+          "description": "Array of VOD's info created by the Bridge",
+          "type": "array",
+          "minItems": 0,
+          "uniqueItems": true,
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/vodentry"
+          }
+        }
```

Comment:
Do we also need a "required" tag here?

Reply:
Yes

Reply:
Added in latest version, please check.

Reply:
Is it the case that only "vods" is marked as required since it is exposed by both oic.if.r and oic.if.baseline?  The other Core Properties may not be exposed by oic.if.r.

Reply:
what is the purpose of vodlist.id ?